### PR TITLE
[REF] Reorganize package to incorporate Transformers and Estimators

### PR DESCRIPTION
**Pull request by tsalo**
References #66.

Changes proposed in this pull request:
- ~~Replace estimator method `fit` with `predict`.~~
- Move stats, io, and utils into new submodule nimare.utils.
- Further consolidate base classes in nimare.base.
- Consolidate Dataset and Database classes.
- Add basic dataset-querying functions (`get_labels`, `get_studies_by_label`, `get_studies_by_mask`, and `get_studies_by_coordinate`).
- Load image files into DataFrame within Dataset the same way labels and coordinates are loaded.

### CODE REVIEW

**Comment by tyarkoni** [comment]
What's the rationale for replacing `fit` with `predict`? I think the semantics of what most of these methods are doing is much closer to fitting than to predicting...

**Comment by tsalo** [comment]
I don't remember the exact reasoning, since this is a cleaned up version of [some refactoring](https://github.com/tsalo/NiMARE/commit/359b69975db9f35e427907c27866862e09e7ecac) I started at Coastal Coding and it's been a little while, but I think I switched to `predict` to better match `sklearn`'s Estimators. Estimators' `fit` method updates the Estimator in place, I think, while `predict` returns a result. I can switch it back to `fit` pretty easily, though.

**Comment by tyarkoni** [comment]
I think we should probably stick with `fit`. It's true that it won't perfectly parallel the sklearn API, but that reflects an underlying divergence in usage patterns. In our case we're almost always going to be returning parameter estimates/summaries from a fitted model, and there will almost never be any prediction per se. So, at least in this respect, statsmodels (which also uses `fit` to fit the model and return a result) is probably a better model for us than sklearn.

**Comment by tsalo** [comment]
Sounds good to me. I'll switch back to `fit`.

**Comment by tsalo** [comment]
@tyarkoni Do you mind if I merge this PR now or would you prefer to look over it first? Behavior should be pretty much the same (except for the consolidation of Database and Dataset classes into a single Dataset class), and tests are passing, so I don't think there are any issues with it except perhaps more high-level conceptual concerns.

**Comment by tyarkoni** [comment]
Fine with me! FWIW, I just skimmed through very quickly and nothing jumped out at me as obviously problematic.
